src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1017,6 +1017,13 @@ void Pane::_ControlConnectionStateChangedHandler(const winrt::Windows::Foundatio
         return;
     }
 
+    if (_overrideCloseOnExit)
+    {
+        // We have been told to close upon process termination regardless of the value of
+        // the profile's CloseOnExit, so go ahead and close
+        Close();

```

Comment:
~couldn't we just like, add another `((mode == CloseOnExitMode::GracefulIfLaunchedByTerminal) ||`~

Nah okay yea I get why we can't do that. We only want to enforce that for defterm panes.

Reply:
So then why not store `_isDefTermSession`/`_isHandoffSession` instead of `_overrideCloseOnExit`? That seems more extensible than having to create an `_override<setting-name>` for each setting we add on like this.

Then you can probably remove/simplify `FinalizeConfigurationGivenDefault`.

Reply:
Renamed `_overrideCloseOnExit` to `_isDefTermSession`, but I would like to keep `FinalizeConfigurationGivenDefault` because I like having a function whose purpose is "making the necessary changes given we're default", which for now is just setting a flag, but I could imagine a future where it needs to do some things more complex than just that (kinda like `FinalizeInheritance` in `Profile`